api/envoy/api/v2/discovery.proto

```diff
@@ -119,55 +119,65 @@ message DiscoveryResponse {
   core.Node node = 1;
 
   // Type of the resource that is being requested, e.g.
-  // "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment". This is implicit
-  // in requests made via singleton xDS APIs such as CDS, LDS, etc. but is
-  // required for ADS.
+  // "type.googleapis.com/envoy.api.v2.ClusterLoadAssignment".
   string type_url = 2;
 
   // DeltaDiscoveryRequests allow the client to add or remove individual
   // resources to the set of tracked resources in the context of a stream.
   // All resource names in the resource_names_subscribe list are added to the
   // set of tracked resources and all resource names in the resource_names_unsubscribe
   // list are removed from the set of tracked resources.
-  // Unlike in state-of-the-world xDS, an empty resource_names_subscribe or
+  //
+  // *Unlike* state-of-the-world xDS, an empty resource_names_subscribe or
   // resource_names_unsubscribe list simply means that no resources are to be
   // added or removed to the resource list.
-  // The xDS server must send updates for all tracked resources but can also
-  // send updates for resources the client has not subscribed to. This behavior
-  // is similar to state-of-the-world xDS.
-  // These two fields can be set for all types of DeltaDiscoveryRequests
-  // (initial, ACK/NACK or spontaneous).
+  // *Like* state-of-the-world xDS, the server must send updates for all tracked
+  // resources, but can also send updates for resources the client has not subscribed to.
   //
   // NOTE: the server must respond with all resources listed in resource_names_subscribe,
   // even if it believes the client has the most recent version of them. The reason:
   // the client may have dropped them, but then regained interest before it had a chance
   // to send the unsubscribe message. See DeltaSubscriptionStateTest.RemoveThenAdd.
   //
+  // These two fields can be set in any DeltaDiscoveryRequest, including ACKs
+  // and initial_resource_versions.
+  //
   // A list of Resource names to add to the list of tracked resources.
   repeated string resource_names_subscribe = 3;
 
   // A list of Resource names to remove from the list of tracked resources.
   repeated string resource_names_unsubscribe = 4;
 
-  // This map must be populated when the DeltaDiscoveryRequest is the
-  // first in a stream (assuming there are any resources - this field's purpose is to enable
-  // a session to continue in a reconnected gRPC stream, and so will not be used in the very
-  // first stream of a session). The keys are the resources names of the xDS resources
-  // known to the xDS client. The values in the map are the associated resource
-  // level version info.
+  // Informs the server of the versions of the resources the xDS client knows of, to enable an xDS
+  // session to continue in a reconnected gRPC stream.
+  // It will not be populated:
+  // * in the very first stream of a session, since the client will not yet have any resources.
```

Comment:
What do you mean by session here? This is probably going to make things a bit more confusing unless we make these terms clear.

Reply:
It means the client's overall conversation with an xDS server(s), spanning stream reconnections. I agree that it's unclear, given that it's not a concept that needs to be discussed very often, and so hasn't gotten an official name.

This 'session' terminology is already used in the text I'm changing, but maybe it could be clarified by calling it "xDS client's session"? My changes in this comment start out by saying the field's purpose is "to enable an xDS session to continue in a reconnected gRPC stream", so maybe making that usage more explicitly defined will do it.